src/geometry.ts

```diff
@@ -44,68 +47,44 @@ export function hexToScreen(x: number, y: number): Point {
     return { x: sx, y: sy }
 }
 
-export function hexFromScreen(x: number, y: number): Point {
-    var x0 = 4800
-    var y0 = 0
-    var nx, ny
-
-    if (x - x0 < 0) nx = (x - x0 + 1) / 16 - 1
-    else nx = (x - x0) / 16
-
-    if (y - y0 < 0) ny = (y - y0 + 1) / 12 - 1
-    else ny = (y - y0) / 12
-
-    if (Math.abs(nx) % 2 != Math.abs(ny) % 2) nx--
-
-    var xhBase = x0 + 16 * nx
-    var yhBase = y0 + 12 * ny
-
-    var hx = (4 * (yhBase - y0) - 3 * (xhBase - x0)) / 96
-    var hy = (yhBase - y0) / 12 - hx / 2
-
-    var dx = x - xhBase
-    var dy = y - yhBase
-
-    // XXX: Some of these cases fall through, should be inspected.
-
-    switch (dy) {
-        case 0:
-            if (dx < 12) {
-                hy--
-                break
-            }
-            if (dx > 18) {
-                if (hx % 2 == 1) hy--
-                hx--
-                break
-            }
-
-        case 1:
-            if (dx < 8) {
-                hx--
-                break
-            }
-            if (dx > 23) {
-                if (hx % 2 == 1) hy--
-                hx--
-                break
-            }
-
-        case 2:
-            if (dx < 4) {
-                hy--
-                break
-            }
-            if (dx > 28) {
-                if (hx % 2 == 1) hy--
-                hx--
-                break
-            }
-        default:
-            break
-    }
+export function pixel2Cube(point: Point): Point3 {
```

Comment:
Please add a doc string that clarifies what this function is doing. Also, I'd prefer `pixelToCube` naming.